Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Staging/Processes/Subsequences/ReduceDemand.mo

```diff
@@ -0,0 +1,341 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Staging.Processes.Subsequences;
+block ReduceDemand "Sequence for reducing operating chiller demand"
+
+  parameter Integer nChi "Total number of chillers in the plant";
+  parameter Real chiDemRedFac = 0.75
+    "Demand reducing factor of current operating chillers";
+  parameter Modelica.SIunits.Time holChiDemTim = 300
+    "Maximum time to wait for the actual demand less than percentage of currnet load";
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uDemLim
+    "Demand limit: true=limit chiller demand"
+    annotation (Placement(transformation(extent={{-200,140},{-160,180}}),
+      iconTransformation(extent={{-140,70},{-100,110}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uChiLoa[nChi](
+    final quantity=fill("HeatFlowRate", nChi),
+    final unit=fill("W", nChi)) "Current chiller load"
+    annotation (Placement(transformation(extent={{-200,110},{-160,150}}),
+      iconTransformation(extent={{-140,30},{-100,70}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput minOPLR(
+    final min=0,
+    final max=1,
+    final unit="1")
+    "Current stage minimum cycling operative partial load ratio"
+    annotation (Placement(transformation(extent={{-200,20},{-160,60}}),
+      iconTransformation(extent={{-140,-10},{-100,30}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uStaDow "Stage down status: true=stage-down"
```

Comment:
The `uDemLim` command will be `true` during the entire staging up process while in the staging-down process, the `uDemLim` will be true only in the first step `DownStart`. By using both `uStaDow` and `uDemLim` in this block would be more clear.



In down process top level controller, it didn't use the `ReduceDemand` block directly, but in fact, it was used in the `DownStart` block which is the first step of the down process (it may simply reduce the demand if the stage-down process does not require chiller on/off, or it requires other process in the first step.)